src/tool/hpcrun/thread_data.c

```diff
@@ -200,6 +200,31 @@ hpcrun_threaded_data(void)
 }
 
 
+void
+hpcrun_thread_init_mem_pool_once(void){
+  static bool is_initialized = false;
```

Comment:
I'm confused.

If the function is called by multiple threads, why isn't `is_initialized` a `pthread_once` variable. 

If `is_initialized` is supposed to be used individually by each thread, shouldn't `is_initialized` be a thread local variable?

Reply:
yes you are right, I will change that